mempool/src/shared_mempool/types.rs

```diff
@@ -75,11 +76,12 @@ impl<V: TransactionValidation + 'static> SharedMempool<V> {
             db,
             validator,
             subscribers,
+            broadcast_within_validator_network: Arc::new(RwLock::new(true)),
         }
     }
 
     pub fn broadcast_within_validator_network(&self) -> bool {
-        self.config.shared_mempool_validator_broadcast
+        *self.broadcast_within_validator_network.read()
```

Comment:
This is essentially the logic change. Instead of taking a config value, take the onchain config value.

This should not be too expensive, because it's only read on the validators which receive broadcasted transactions in batches.

Reply:
It's well-behaved going from `true` to `false`, but the other way around we can have transactions that are in mempool and are never broadcasted or in a batch.

We really don't want to go back from quorum store, so I think it's ok to have this disruption. WDYT?

I can add this as a comment if it makes sense.